### PR TITLE
[cmake] Small improvements and fixes

### PR DESCRIPTION
**Pull request by fetzerch**
See commit messages for details.

Pings:
@laurimyllari, @FernetMenta for the build fix 0789567. This ensures that ColorManager can be built on Windows, but also for Linux I think it's better not to rely on non-standard techniques. Would be great if you could test if I didn't break anything, I've just compile tested it here.

@Fneufneu: 48993cc fixes the in-source builds, as discussed in https://github.com/xbmc/xbmc/commit/cc735b78888d58cf8bf8608905ae315782e21ed0#commitcomment-18124684

For the rest, maybe @hudokkow want's to have a quick look :)

### CODE REVIEW

File: project/cmake/scripts/common/CMakeHelpers.cmake [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+# Print all CMake variables.
+macro(debug_print_variables)
+  get_cmake_property(_variableNames VARIABLES)
+  foreach(_variableName ${_variableNames})
+    message(STATUS "${_variableName} = ${${_variableName}}")
+  endforeach()
+endmacro()
+
+# Get all propreties that cmake supports and convert them to a list.
```
Comment (hudokkow) [inline project/cmake/scripts/common/CMakeHelpers.cmake]: s/propreties/properties
Rest looks fine. I had to pick on something, :stuck_out_tongue_winking_eye:

**Comment by FernetMenta** [comment]
@fetzerch thanks! Sorry, I did not notice the non-standard on review.

**Comment by fetzerch** [comment]
@hudokkow: Fixed 2 typos, thanks.

**Comment by fetzerch** [comment]
jenkins build and merge

**Pull request merged**